hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileSystemDeleteHandler.java

```diff
@@ -63,14 +63,17 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
 
         final var transactionBody = context.body().systemDeleteOrThrow();
         final var fileStore = context.createStore(ReadableFileStoreImpl.class);
-        final var fileMeta = preValidate(transactionBody.fileID(), fileStore, context, true);
+        preValidate(transactionBody.fileID(), fileStore, context, true);
 
-        validateAndAddRequiredKeys(fileMeta.keys(), context, true);
+        var file = fileStore.getFileLeaf(transactionBody.fileID());
+        validateAndAddRequiredKeys(file.orElse(null), null, context);
     }
 
     @Override
     public void handle(@NonNull final HandleContext handleContext) throws HandleException {
         requireNonNull(handleContext);
+        // First validate this file is mutable; and the pending mutations are allowed
+        // TODO: add or condition for privilege accounts from context
```

Comment:
Might want to revise the comment:
```suggestion
        // TODO: check here that the "payer" is a privileged account.
        //       a privileged account is always required for this transaction.
```

Reply:
done thanks